### DISCUSSION TITLE
Expected behavior for devices with running scripts

### DISCUSSION THREAD

**Discussion by jakobrosenberg** [pycom/pymakr-vsc#227] [Q&A]
For a device that's running the LED example:
- What is the expected behavior when I connect (lightning icon) to it in Pymakr?
- What is the expected behavior if I hard reset the device from Pymakr (this starts the scripts) and then click the "open terminal" icon?

<details>
<summary>LED Example</summary>

```python
import pycom
import time

pycom.heartbeat(False)

while True:
    pycom.rgbled(0xFF0000)  # Red
    time.sleep(1)
    pycom.rgbled(0x00FF00)  # Green
    time.sleep(1)
    pycom.rgbled(0x0000FF)  # Blue
    time.sleep(1)
```

</details>

**Comment by Josverl** [comment]
IMO **by default:** 
- connecting to the device should not interrupt a running script on the repl. ( ie do not send Ctrl-C , Ctrl-C) 
Expected: the leds to continue to blink

  - I do think it makes sense to have a per project/device option to enable such an interrupt


 - after a hard reset (how is this implemented, machine.reset() or Ctrl-D ?) , 
   - for a serial connection:  I expect the connection to continue, and the terminal/repl to resume the same state that it was
   - for a network connection: nice to have: try to re-establish the network connection ( up to a timeout) 
  
TLDR: Expected: the leds to continue to blink
